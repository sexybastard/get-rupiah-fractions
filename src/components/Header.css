.title-container {
  text-align: center;
  margin-bottom: 20px;
  border-bottom: solid 1px #333;
}

.title-container h1 {
  padding-bottom: 20px;
  color: #333;
}