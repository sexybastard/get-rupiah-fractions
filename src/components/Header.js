import React from 'react';
import './Header.css';

const defaultTitle = 'Default Title'

const Header = ({ title }) => {
  const getTitle = title ? title : defaultTitle

  return (
    <div id="title-container" className="title-container">
      <h1>{getTitle}</h1>
    </div>
  )
}

export default Header;
