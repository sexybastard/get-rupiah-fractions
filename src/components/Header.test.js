import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe("Component: Header should match snapshoot", () => {
  it("Case with props title set", () => {
    const title = 'Sample title'
    let wrapper = shallow(<Header title={title} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("Case with no props", () => {
    let wrapper = shallow(<Header />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
})

describe("Component: Header should have the correct className and content", () => {
  it("Ensure to have a correct className", () => {
    const title = 'Sample title'
    const wrapper = shallow(<Header title={title} />);
    expect(wrapper.find('#title-container').prop('className')).toEqual('title-container')
  });

  it("Ensure to have a correct content text", () => {
    const title = 'Sample title'
    let wrapper = shallow(<Header title={title} />);
    expect(wrapper.find('#title-container h1').text()).toEqual(title)
    wrapper = shallow(<Header />);
    expect(wrapper.find('#title-container h1').text()).toEqual('Default Title')
  });
});
