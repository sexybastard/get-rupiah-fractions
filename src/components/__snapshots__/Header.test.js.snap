// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should match snapshoot Case with no props 1`] = `
Array [
  <div
    className="title-container"
    id="title-container"
  >
    <h1>
      Default Title
    </h1>
  </div>,
]
`;

exports[`Component: Header should match snapshoot Case with props title set 1`] = `
Array [
  <div
    className="title-container"
    id="title-container"
  >
    <h1>
      Sample title
    </h1>
  </div>,
]
`;
